pymc_marketing/mmm/base.py

```diff
@@ -404,6 +420,97 @@ def plot_posterior_predictive(
             raise RuntimeError("The model hasn't been fit yet, call .fit() first")
         return fig
 
+    def plot_errors(
+        self, original_scale: bool = False, ax: plt.Axes = None, **plt_kwargs: Any
+    ) -> plt.Figure:
+        """Plot model errors by taking the difference between true values and predicted.
+
+        Parameters
+        ----------
+        original_scale : bool, optional
+            Whether to plot in the original scale.
+        ax : plt.Axes, optional
+            Matplotlib axis object.
+
+        Returns
+        -------
+        plt.Figure
+        """
+        posterior_predictive_data: Dataset = self.posterior_predictive
+
+        target = np.asarray(
+            transform_1d_array(self.get_target_transformer().transform, self.y)
+        )
+
+        if len(target) != len(posterior_predictive_data.date):
+            raise ValueError(
+                "The length of the target variable doesn't match the length of the date column. "
+                "If you are computing out-of-sample errors, please overwrite `self.y` with the "
+                "corresponding (non-transformed) target variable."
+            )
+
+        target_broadcast = np.atleast_1d(target)[np.newaxis, np.newaxis, ...]
+        errors = target_broadcast - posterior_predictive_data
```

Comment:
Might be easier to just use the xarray broadcasting here since posterior_predictive is Dataset

target = pd.Series(target, index=posterior_predictive_data.date).to_xarray() # ?
# Then broadcasting comes along for free
errors = target - posterior_predictive_data